### ISSUE TITLE
nivoslider.jquery.json' is not proper Cucumber report!

### ISSUE THREAD

**Issue by rahulkulkarniyahoo** [jenkinsci/cucumber-reports-plugin#174]
hi all,

Can you please help me sort out issue which I encountered.
Versions:
Cucumber Plugin Version: 3.9.0
Jenkins Version: Jenkins ver. 2.62

I have included the cucmber.json and nivoslider.jquery.json files in text format.
I received the following error:
*********************** ERROR **********************************

net.masterthought.cucumber.ValidationException: File '/XXXX/jenkins/jobs/NativeApp_iOS_Regression/builds/7/cucumber-html-reports/.cache/site/serenity/nivo-slider/3.2/nivoslider.jquery.json' is not proper Cucumber report!
at net.masterthought.cucumber.ReportParser.parseForFeature(ReportParser.java:87)
at net.masterthought.cucumber.ReportParser.parseJsonFiles(ReportParser.java:58)
at net.masterthought.cucumber.ReportBuilder.generateReports(ReportBuilder.java:87)
at net.masterthought.jenkins.CucumberReportPublisher.generateReport(CucumberReportPublisher.java:269)
at net.masterthought.jenkins.CucumberReportPublisher.perform(CucumberReportPublisher.java:207)
at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:81)
at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:730)
at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:676)
at hudson.maven.MavenModuleSetBuild$MavenModuleSetBuildExecution.post2(MavenModuleSetBuild.java:1072)
at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:621)
at hudson.model.Run.execute(Run.java:1760)
at hudson.maven.MavenModuleSetBuild.run(MavenModuleSetBuild.java:542)
at hudson.model.ResourceController.execute(ResourceController.java:97)
at hudson.model.Executor.run(Executor.java:415)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of net.masterthought.cucumber.json.Feature[] out of START_OBJECT token
at [Source: java.io.InputStreamReader@864a7cb; line: 1, column: 1]
at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:270)
at com.fasterxml.jackson.databind.DeserializationContext.reportMappingException(DeserializationContext.java:1234)
at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1122)
at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1075)
at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.handleNonArray(ObjectArrayDeserializer.java:275)
at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:179)
at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:20)
at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3798)
at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2887)
at net.masterthought.cucumber.ReportParser.parseForFeature(ReportParser.java:81)
... 14 more

********************** NIVO SLIDER jquery.json **********************

{
"name": "nivoslider",
"version": "3.2",
"title": "Nivo Slider",
"author": {
"name": "Gilbert Pellegrom",
"url": "http://gilbert.pellegrom.me"
},
"licenses": [
{
"type": "MIT",
"url": "https://github.com/gilbitron/Nivo-Slider/blob/master/license.txt"
}
],
"dependencies": {
"jquery": ">=1.7"
},
"description": "The Most Awesome jQuery Image Slider.",
"keywords": [
"slider",
"animation",
"effect"
],
"homepage": "http://nivo.dev7studios.com",
"docs": "http://nivo.dev7studios.com/support",
"demo": "http://nivo.dev7studios.com/demos",
"download": "http://nivo.dev7studios.com/pricing",
"bugs": "https://github.com/gilbitron/Nivo-Slider/issues"
}

********************** JSON FILE *******************************

[
{
"line": 1,
"elements": [
{
"before": [
{
"result": {
"duration": 32102794905,
"status": "passed"
},
"match": {
"location": "StepDefinitions.startUp()"
}
}
],
"line": 36,
"name": "Verify Guest Profile login and Select Police Station appears on Map @lancaster",
"description": "",
"id": "login-and-switching-profiles-for-lancaster;verify-guest-profile-login-and-select-police-station-appears-on-map-@lancaster;;2",
"after": [
{
"result": {
"duration": 669218730,
"status": "passed"
},
"match": {
"location": "StepDefinitions.tearDown()"
}
}
],
"type": "scenario",
"keyword": "Scenario Outline",
"steps": [
{
"result": {
"duration": 105139080,
"status": "passed"
},
"line": 27,
"name": "I am on Lancaster App",
"match": {
"location": "StepDefinitions.i_am_on_Lancaster_App()"
},
"keyword": "Given "
},
{
"result": {
"duration": 9488782143,
"status": "passed"
},
"line": 28,
"name": "I select profile "City, Coast \u0026 Countryside"",
"match": {
"arguments": [
{
"val": "City, Coast \u0026 Countryside",
"offset": 18
}
],
"location": "StepDefinitions.i_select_profile(String)"
},
"matchedColumns": [
0
],
"keyword": "When "
},
{
"result": {
"duration": 6552287484,
"status": "passed"
},
"line": 29,
"name": "I am able to login",
"match": {
"location": "StepDefinitions.i_am_able_to_login()"
},
"keyword": "Then "
},
{
"result": {
"duration": 10749743595,
"status": "passed"
},
"line": 30,
"name": "I can click on Map tile",
"match": {
"location": "StepDefinitions.i_can_click_on_Map_tile()"
},
"keyword": "And "
},
{
"result": {
"duration": 3813902194,
"status": "passed"
},
"line": 31,
"name": "I can search for "Police Stations (2)"",
"match": {
"arguments": [
{
"val": "Police Stations (2)",
"offset": 18
}
],
"location": "StepDefinitions.i_can_search_for(String)"
},
"matchedColumns": [
1
],
"keyword": "And "
},
{
"result": {
"duration": 4108974459,
"status": "passed"
},
"line": 32,
"name": "select the "Lancaster Police Station"",
"match": {
"arguments": [
{
"val": "Lancaster Police Station",
"offset": 12
}
],
"location": "StepDefinitions.select_the(String)"
},
"matchedColumns": [
2
],
"keyword": "And "
},
{
"result": {
"duration": 17091734828,
"status": "passed"
},
"line": 33,
"name": "I can confirm the Map Pin exists with "Lancaster Police Station"",
"match": {
"arguments": [
{
"val": "Lancaster Police Station",
"offset": 39
}
],
"location": "StepDefinitions.i_can_confirm_the_Map_Pin_exists_with(String)"
},
"matchedColumns": [
2
],
"keyword": "And "
}
],
"tags": [
{
"line": 25,
"name": "@guest"
},
{
"line": 25,
"name": "@smoke"
},
{
"line": 25,
"name": "@lancaster"
},
{
"line": 25,
"name": "@Regression"
},
{
"line": 25,
"name": "@test"
}
]
}
],
"name": "Login and Switching Profiles for Lancaster",
"description": "",
"id": "login-and-switching-profiles-for-lancaster",
"keyword": "Feature",
"uri": "Lancaster.feature"
}
]

**Comment by rahulkulkarniyahoo** [comment]
This issue was fixed by adding to File Include Pattern: **/cucumber.json

The default pattern **/*.json was picking up the nivoslider.jquery.json from another folder causing the issue. 

I had tried to restrict by defining the Report Path field, e.g. 
Report Path field: {$workspace}/target/
File Include Pattern: **/*.json
But even after this, the nivoslider.jquery.json from another folder and not was getting identified causing the error.

If I provided the path to the location of cucumber.json file, why was the plugin still trying to use that folder?
